main.nf

```diff
@@ -1,14 +1,179 @@
+      .multiMap { id, reads, bai -> 
+        reads: tuple(id, reads)
+        bai: bai
+      }
+      .set { bwa_mem_dedup_ch }
+    bwa_mem_dedup(bwa_mem_dedup_ch.reads, bwa_mem_dedup_ch.bai)
+    samtools_index_assembly(bwa_mem_dedup.out.bam)
+
+    // construct freebayes input channels
+    assembly
+      .join(bwa_mem_dedup.out.bam)
+      .join(samtools_index_assembly.out.bai)
+      .multiMap { id, fasta, bam, bai -> 
+        assembly: tuple(id, fasta)
+        mapping: tuple(bam, bai)
+      }
+      .set { freebayes_ch }
+
+    freebayes(freebayes_ch.assembly, freebayes_ch.mapping)
+    mask_polymorph_assembly(assembly.join(freebayes.out.vcf))
+    quast(assembly, genomeReference)
+    mlst(assembly, params.species, mlstDb)
+    // split assemblies and id into two seperate channels to enable re-pairing
+    // of results and id at a later stage. This to allow batch cgmlst/wgmlst analysis 
+    mask_polymorph_assembly.out.fasta
+      .multiMap { sampleName, filePath -> 
+        sampleName: sampleName
+        filePath: filePath
+      }
+      .set{ maskedAssemblyMap }
+
+    chewbbaca_create_batch_list(maskedAssemblyMap.filePath.collect())
+    chewbbaca_allelecall(maskedAssemblyMap.sampleName.collect(), chewbbaca_create_batch_list.out.list, chewbbacaDb, trainingFile)
+    chewbbaca_split_results(chewbbaca_allelecall.out.sampleName, chewbbaca_allelecall.out.calls)
+
+    // end point
+    export_to_cdm(chewbbaca_split_results.out.output.join(quast.out.qc).join(post_align_qc.out.qc))
+
+    // sourmash
+    sourmash(assembly)
+
+    // antimicrobial detection (amrfinderplus & abritamr)
+    amrfinderplus(assembly, amrfinderDb)
+    //abritamr(amrfinderplus.out.output)
+
+    // perform resistance prediction
+    resfinder(reads, params.species, resfinderDb, pointfinderDb)
+    virulencefinder(reads, params.useVirulenceDbs, virulencefinderDb)
+
+    // combine results for export
+    quast.out.qc
+      .join(post_align_qc.out.qc)
+      .join(mlst.out.json)
+      .join(chewbbaca_split_results.out.output)
+      .join(amrfinderplus.out.output)
+      .join(resfinder.out.json)
+      .join(resfinder.out.meta)
+      .join(virulencefinder.out.json)
+      .join(virulencefinder.out.meta)
+      .join(save_analysis_metadata.out.meta)
+      .set{ combinedOutput }
+
+    // Using kraken for species identificaiton
+    if ( params.useKraken ) {
+      krakenDb = file(params.krakenDb, checkIfExists: true)
+      kraken(reads, krakenDb)
+      bracken(kraken.out.report, krakenDb).output
+      combinedOutput = combinedOutput.join(bracken.out.output)
+      create_analysis_result(combinedOutput)
+	  } else {
+      emptyBrackenOutput = reads.map { sampleName, reads -> [ sampleName, [] ] }
+      combinedOutput = combinedOutput.join(emptyBrackenOutput)
+      create_analysis_result(combinedOutput)
+	  }
+    
+  emit: 
+    pipeline_result = create_analysis_result.output
+    cdm_import = export_to_cdm.output
+=======
```

Comment:
Everythin above and including this line should be removed as it is in the workflows.